docs/designers-developers/developers/feature-flags.md

```diff
@@ -0,0 +1,114 @@
+# Feature Flags
+
+With phase 2 of the Gutenberg project there's a need for improved control over how code changes are released. Newer features developed for phase 2 and beyond should only be released to the Gutenberg plugin, while improvements and bug fixes should still continue to make their way into core releases.
+
+The technique for handling this is known as a 'feature flag'. 
+
+## Introducing `window.GUTENBERG_PHASE`
```

Comment:
While it's quite thorough, I think most people coming to this document care most about the **Basic Use** section, so at the least I might suggest moving it to precede this section, if not simplify / cut parts of this section.